src/engine/app.rs

```diff
@@ -16,11 +16,10 @@ pub struct Application {
 }
 
 impl Application {
-    /// Creates a new Application with the given initial game state, planner, and config.
-    pub fn new<T>(initial_state: T, planner: Planner<Arc<Mutex<Context>>>, config: Config) -> Application
+    /// Creates a new Application with the given initial game state, planner, and context.
+    pub fn new<T>(initial_state: T, planner: Planner<Arc<Mutex<Context>>>, context: Context) -> Application
```

Comment:
Didn't notice this line before. Why do we pass in two Contexts into this method exactly, one enclosed in a Planner and one alone? Can't the `new()` method build a Planner itself, insert the given State and Context inside, and pass it into the StateMachine?